### PR TITLE
Enable provider users to be removed via Support

### PR DESCRIPTION
**Pull request by duncanjbrown**
## Context

When we moved to the new provider user flow we lost the ability to revoke Providers' access via Support

## Changes proposed in this pull request

_First soup up the SummaryListComponent_

- support multiple actions
- enable us to add a `data-qa` field by specifying it on the row, to simplify testing

_Then add new flow_

### "Remove access" link

Screenshot 2021-07-06 at 17 45 11

### Confirmation page

Screenshot 2021-07-06 at 17 45 17

### Flash message

Screenshot 2021-07-06 at 17 45 22

### Empty state

Screenshot 2021-07-06 at 17 45 32

## Link to Trello card

https://trello.com/c/wjbc7uEt/3952-add-remove-links-to-new-provider-user-flow-in-support

## Things to check

- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -919,6 +919,8 @@
       patch '/permissions' => 'single_provider_users#update', as: :update_permissions
       get '/notifications/edit' => 'single_provider_user_notifications#edit', as: :edit_provider_notifications
       put '/notifications' => 'single_provider_user_notifications#update', as: :update_provider_notifications
+      get '/permissions/:provider_permissions_id/remove' => 'single_provider_user_removals#new', as: :provider_user_removals
+      post '/permissions/:provider_permissions_id/remove' => 'single_provider_user_removals#create', as: :remove_provider_user
```
Comment (cwrw) [inline config/routes.rb:923]: Maybe we can use a delete/destroy path as it's more representative of what is happening here?

File: app/views/support_interface/single_provider_user_removals/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<span class="govuk-caption-xl"><%= @user.display_name %></span>
```
Comment (cwrw) [inline app/views/support_interface/single_provider_user_removals/new.html.erb:1]: can we also add a back link, just to inline with all our other designs?

**Review by cwrw — COMMENTED** [review]
When removing all permissions for a user and navigating to https://apply-for-te-3952-add-r-ms4v7r.herokuapp.com/support/users/provider, and attempting to update the permissions for  that user we are getting an empty page, example: https://apply-for-te-3952-add-r-ms4v7r.herokuapp.com/support/providers/users/1/permissions/edit

Maybe we should hide that button

**Comment by duncanjbrown** [comment]
Thanks for review @cwrw! Good point, I forgot we had the link on the index page too. I'll remove that and change the route.

File: app/views/support_interface/single_provider_user_removals/new.html.erb (hunk 2 above)
Comment (duncanjbrown) [reply app/views/support_interface/single_provider_user_removals/new.html.erb:1]: good call, done

**Comment by duncanjbrown** [comment]
Hopefully ready to go!

**Review by cwrw — APPROVED** [review]
Thanks for addressing all teh comments. Just final thing around changing `create` to `destroy` otherwise 👍

**Pull request merged**